ding/utils/traffic_helper.py

```diff
@@ -0,0 +1,158 @@
+from collections import deque
+from typing import TYPE_CHECKING
+import time
+import os
+import json
+import logging
+from random import random
+import pandas as pd
+import ding.utils
+
+if TYPE_CHECKING:
+    from ding.framework import Parallel
+
+
+class Traffic:
+
+    def __init__(self) -> None:
+        pass
+
+    def config(
+            self,
+            file_path: str = None,
+            online: bool = False,
+            maxlen: int = 10000,
+            router: "Parallel" = None
+    ) -> "Traffic":
+        """
+        Overview:
+            Enable and change the configuration of a Traffic instance. 
+        Arguments:
+            - file_path (:obj:`str`): File path that offline data are to be saved at. \
+                In local mode, it must be set sucessfully for at least once. \
+                In remote mode, no need to set it for worker, but is necessary for master. \
+            - online (:obj:`bool`): To enable online analysis and maintain data in memory.
+            - maxlen (:obj:`int`): Max data size in memory.
+            - router (:obj:`Parallel`): To enable remote mode.
+        """
+
+        if online and maxlen <= 0:
+            logging.error("Illegal data size.")
+            return self
+
+        if not hasattr(self, "_file") or self._file is None:
+            self._file = None
+            self._has_file_writer = False
+            self._file_path = None
```

Comment:
Why do you need so many properties for one feature?

Reply:
Variables are reduced.